fury/utils.py

```diff
@@ -170,23 +172,17 @@ def numpy_to_vtk_image_data(array, spacing=(1.0, 1.0, 1.0),
     vtk_image : vtkImageData
 
     """
-    if array.ndim not in [2, 3]:
-        raise IOError("only 2D (L, RGB, RGBA) or 3D image available")
+    if array.ndim > 2:
+        raise IOError("Array Dimensionality Should be 2 or less!")
 
     vtk_image = ImageData()
-    depth = 1 if array.ndim == 2 else array.shape[2]
 
-    vtk_image.SetDimensions(array.shape[1], array.shape[0], depth)
-    vtk_image.SetExtent(0, array.shape[1] - 1,
-                        0, array.shape[0] - 1,
-                        0, 0)
+    vtk_image.SetDimensions(dimension)
     vtk_image.SetSpacing(spacing)
     vtk_image.SetOrigin(origin)
-    temp_arr = np.flipud(array)
-    temp_arr = temp_arr.reshape(array.shape[1] * array.shape[0], depth)
-    temp_arr = np.ascontiguousarray(temp_arr, dtype=array.dtype)
+
     vtk_array_type = numpy_support.get_vtk_array_type(array.dtype)
-    uchar_array = numpy_support.numpy_to_vtk(temp_arr, deep=deep,
+    uchar_array = numpy_support.numpy_to_vtk(array, deep=deep,
                                              array_type=vtk_array_type)
     vtk_image.GetPointData().SetScalars(uchar_array)
     return vtk_image
```

Comment:
I am not sure what you are doing here. Why only 2D image?

the previous code is good so I do not understand why you modify it and add dimension...

Reply:
When working with the `actor.slicer` and `actor.contour_from_roi` we are getting 1D array and in other cases we are getting 2D arrays so to make the function more generic I kept all the array conversion logic outside the function due to which I changed the if condition.
Similarly each method was setting different dimension for the ImageData so made that as a property.

Reply:
I am not sure about this approach, still thinking and checking some stuff. Thank you for your feedback